### PR TITLE
Fix ExpressionEquivalence for long timestamp

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/type/LongTimestamp.java [hunk 1]
```diff
@@ -63,4 +64,14 @@ public int hashCode()
...
     }
+
+    @Override
+    public int compareTo(LongTimestamp other)
+    {
+        int value = Long.compare(epochMicros, other.epochMicros);
+        if (value != 0) {
+            return value;
+        }
+        return Integer.compareUnsigned(picosOfMicro, other.picosOfMicro);
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/type/LongTimestamp.java:75]: no need for unsignedm sine they are in "signed" range

File: presto-main/src/main/java/io/prestosql/sql/planner/optimizations/ExpressionEquivalence.java [hunk 2]
```diff
@@ -285,8 +284,13 @@ public int compare(RowExpression left, RowExpression right)
                 if (javaType == float.class || javaType == double.class) {
                     return Double.compare(((Number) leftValue).doubleValue(), ((Number) rightValue).doubleValue());
                 }
-                if (javaType == Slice.class) {
-                    return ((Slice) leftValue).compareTo((Slice) rightValue);
+                if (leftValue instanceof Comparable) {
+                    try {
+                        //noinspection unchecked
+                        return ((Comparable<Object>) leftValue).compareTo(rightValue);
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/sql/planner/optimizations/ExpressionEquivalence.java:290]: techncally this cast is wrong, `leftValue` is not `Comparable<Object>`
i'd rather cast to a raw type `Comparable` (and add `rawtypes` to noinspection as well)

File: presto-spi/src/main/java/io/prestosql/spi/type/LongTimestampWithTimeZone.java [hunk 3]
```diff
@@ -66,4 +69,34 @@ public short getTimeZoneKey()
...
+    }
+
+    @Override
+    public int compareTo(LongTimestampWithTimeZone other)
+    {
+        int value = Long.compare(epochMillis, other.epochMillis);
+        if (value != 0) {
+            return value;
+        }
+        return Integer.compareUnsigned(picosOfMilli, other.picosOfMilli);
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/type/LongTimestampWithTimeZone.java:100]: `Integer.compare`

File: presto-spi/src/main/java/io/prestosql/spi/type/LongTimestamp.java (hunk 1 above)
Comment (dain) [reply presto-spi/src/main/java/io/prestosql/spi/type/LongTimestamp.java:75]: This matches the compare implementation in the Type class.

File: presto-spi/src/main/java/io/prestosql/spi/type/LongTimestampWithTimeZone.java (hunk 3 above)
Comment (dain) [reply presto-spi/src/main/java/io/prestosql/spi/type/LongTimestampWithTimeZone.java:100]: This matches the compare implementation in the Type class.

**Pull request merged**